src/inputs/ChipSelectField.test.tsx

```diff
@@ -0,0 +1,133 @@
+  it("displays custom placeholder", async () => {
+    // Given a ChipSelectField without a value and a custom placeholder
+    const r = await render(
+      <TestComponent label="Test Label" value={undefined} options={sports} placeholder="+ Task Status" />,
+    );
+    // Then the custom placeholder is shown
+    expect(r.chipSelectField()).toHaveTextContent("+ Task Status");
+  });
+
+  it("is clearable", async () => {
+    // Given a ChipSelectField that is clearable
+    const onSelect = jest.fn();
+    const r = await render(<TestComponent label="Label" value="s:2" options={sports} clearable onSelect={onSelect} />);
+    // With an existing value
+    expect(r.chipSelectField()).toHaveTextContent("Soccer");
+    // When clicking the clear button
+    click(r.chipSelectField_clearButton);
+    // Then expect the default placeholder value to display
+    expect(r.chipSelectField()).toHaveTextContent("Select an option");
+    // And the clear button is removed when no value is selected
+    expect(r.queryByTestId("chipSelectField_clearButton")).toBeFalsy();
+    // And onSelect to be called
+    expect(onSelect).toBeCalledWith([undefined, undefined]);
+  });
+
+  it("can select options", async () => {
+    // Given a ChipSelectField
+    const onSelect = jest.fn();
+    const r = await render(<TestComponent label="Label" value="s:2" options={sports} onSelect={onSelect} />);
+    // With an existing value
+    expect(r.chipSelectField()).toHaveTextContent("Soccer");
+    // When selecting a new value
+    click(r.chipSelectField);
+    click(r.getByRole("option", { name: "Basketball" }));
+    // Then the field's value updates
+    expect(r.chipSelectField()).toHaveTextContent("Basketball");
+    // And onSelect is called
+    expect(onSelect).toBeCalledWith(["s:3", sports[2]]);
+  });
+
+  it("fires onFocus and onBlur events", async () => {
+    // Given a ChipSelectField
+    const onFocus = jest.fn();
+    const onBlur = jest.fn();
+    const r = await render(
+      <TestComponent label="Label" value="s:2" options={sports} onFocus={onFocus} onBlur={onBlur} />,
+    );
+    // When firing focus and blur events, then expect callbacks to be triggered
+    fireEvent.focus(r.chipSelectField());
+    expect(onFocus).toBeCalledTimes(1);
+    fireEvent.blur(r.chipSelectField());
+    expect(onBlur).toBeCalledTimes(1);
+  });
+
+  it("does not fire onBlur when opening menu and returns focus to the button upon selecting from menu", async () => {
+    // Given a ChipSelectField
+    const onFocus = jest.fn();
+    const onBlur = jest.fn();
+    const r = await render(
+      <TestComponent label="Label" value="s:2" options={sports} onFocus={onFocus} onBlur={onBlur} />,
+    );
+
+    // When focusing the field and opening the menu
+    click(r.chipSelectField);
+    // And onFocus should have been called
+    expect(onFocus).toBeCalledTimes(1);
+    // And when firing the blur event while the menu is opened,
+    fireEvent.blur(r.chipSelectField());
```

Comment:
**question:** This is neat! How are you controlling the blur to not be propagated?

Reply:
We are able to conditionally call on blur based on whether the menu is open or not.

Reply:
Is that through React-Aria?

Reply:
Not aria, but, React-Stately - same idea. We know when the menu is opened/closed.